File: appli-web/src/app/core/api/nutrition-off-api.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { OffProduct } from '@core/models/off-product.model';
import { API_BASE_URL } from './api.config';

/**
 * Client du proxy Open Food Facts serveur (docs/NUTRITION_DESIGN.md §4.1) — read-only,
 * pas une entité synchronisable. Le client copie un produit choisi dans son catalogue
 * `foods` (source=OFF, sourceRef=barcode) via FoodRepository.importFromOff.
 */
@Injectable({ providedIn: 'root' })
export class NutritionOffApi {
  private readonly http = inject(HttpClient);
  private readonly base = `${API_BASE_URL}/nutrition/off`;

  search(q: string, pageSize = 20): Observable<OffProduct[]> {
    const params = new HttpParams().set('q', q).set('pageSize', pageSize);
    return this.http.get<OffProduct[]>(`${this.base}/search`, { params });
  }

  getProduct(barcode: string): Observable<OffProduct> {
    return this.http.get<OffProduct>(`${this.base}/product/${barcode}`);
  }
}

File: appli-android/app/src/main/java/com/example/sportapp/core/network/NutritionOffApi.kt
package com.example.sportapp.core.network

import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Proxy serveur Open Food Facts (Nutrition V2, docs/NUTRITION_DESIGN.md §4.1).
 * Read-only : recherche texte + lookup par code-barres, normalisés per-100 g
 * vers le format [OffProduct] (copié dans le catalogue local à l'import).
 * Auth JWT obligatoire (politique 8) — passe par le client authentifié.
 */
interface NutritionOffApi {

    @GET("nutrition/off/search")
    suspend fun search(
        @Query("q") q: String,
        @Query("pageSize") pageSize: Int = 20,
    ): List<OffProduct>

    @GET("nutrition/off/product/{barcode}")
    suspend fun product(@Path("barcode") barcode: String): OffProduct
}
